src/Datepicker.js

```diff
@@ -68,6 +73,24 @@ export default class Datepicker extends Component {
     })
   }
 
+  handleChange(e) {
+    const text = e.target.value
+    this.setState({
+      textInput: text
+    })
+    const dateRegex = /\d{2}\/\d{2}\/\d{4}/
+
+    // check length first to avoid doing excessive regex checks
+    if(text.length === 10 && dateRegex.test(text)) {
```

Comment:
is there not something like this built into `date-fns`

Reply:
As best I can tell, no--there's nothing to check the actual _format_ of the date string, just the contents.